src/classes/OPP_PrimaryContact_BATCH.cls

```diff
@@ -57,20 +63,21 @@ public class OPP_PrimaryContact_BATCH implements Database.Batchable<sObject> {
         //creates an opportunity object
         //if there are duplicate primary contact roles, take the oldest modified one, so if the
         //duplicate is the result of a merge, the winner of the merge keeps the primary contact role
-        map<id,Opportunity> mapOppsForUpdate = new map<id,Opportunity>();
-        for (OpportunityContactRole ocr : (list<OpportunityContactRole>)scope) {
-            mapOppsForUpdate.put(ocr.OpportunityId, new Opportunity(id=ocr.OpportunityId, Primary_Contact__c=ocr.ContactId));
+        Map<Id, Opportunity> mapOppsForUpdate = new Map<Id, Opportunity>();
```

Comment:
Rather than cast `scope` as `List<OpportunityContactRole>` twice in this method, you can just change the method signature to the following this way `scope` comes in as `List<OpportunityContactRole>` from the start.

```
public void execute(Database.BatchableContext BC, List<OpportunityContactRole> scope)
```